fury/data/fetcher.py

```diff
@@ -75,41 +75,47 @@ def _already_there_msg(folder):
     print(msg)
 
 
-def _get_file_md5(filename):
-    """Compute the md5 checksum of a file."""
-    md5_data = md5()
-    with open(filename, 'rb') as f:
-        for chunk in iter(lambda: f.read(128 * md5_data.block_size), b''):
-            md5_data.update(chunk)
-    return md5_data.hexdigest()
+def _get_file_sha(filename):
+    """Generates SHA checksum for the entire file in blocks of 256
 
+    Parameters
+    ----------
+    filename: str
+        The path to the file whose sha checksum is to be generated
 
-def check_md5(filename, stored_md5=None):
-    """Compute the md5 of filename.
+    """
+    sha256_data = sha256()
+    with open(filename, 'rb') as f:
+        for chunk in iter(lambda: f.read(256*sha256_data.block_size), b''):
+            sha256_data.update(chunk)
+    return sha256_data.hexdigest()
 
-    check if it matches with the supplied string md5
+def check_sha(filename, stored_sha256=None):
+    """Checks the generated sha checksum.
 
     Parameters
     ----------
-    filename : string
-        Path to a file.
-    md5 : string
-        Known md5 of filename to check against. If None (default), checking is
-        skipped
+    filename: str
+        The path to the file whose checksum is to be compared
+    stored_sha: string input, optional
+        Used to verify the generated SHA checksum
```

Comment:
Do not forget this important part in the docstring:  `If None (default), checking is skipped`